### PR TITLE
[admin-tool] Some improvements to execute-data-recovery command

### PR DESCRIPTION
**Pull request by lluwm**
This change does the following improvements to admin-tool execute-data-recovery command:

1. Remove password related codes as passing password plaintext from admin-tool to venice-tools is not an acceptable solution.

2. Support sentinel run in DataRecoveryExecutor. By having a sentinel run (normally requires manually inputting credentials thus creating a local Azkaban session file), the follow-up repush tasks can reuse the local session info for following interactions with Azkaban without the need to input user credentials before the session is expired. So, we only need to type in user credentials once for multiple stores.

   If the sentinel run succeeds, it will continue to run (probably in parallel but depends on the executor pool size configs) for remaining stores. However, if the sentinel run fails, then it stops to run remaining store and reports error message.

3. Add non-interactive support for testing automation purpose in future.

## Does this PR introduce any user-facing changes?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Make sure to explain your proposed changes and call out the behavior change.

### CODE REVIEW

File: clients/venice-admin-tool/src/main/java/com/linkedin/venice/datarecovery/DataRecoveryTask.java [hunk 1]
```diff
@@ -19,6 +19,10 @@ public void run() {
     taskResult = new TaskResult(command.getResult());
   }
 
+  public boolean requiresSentinelRun() {
```
Comment (nisargthakkar) [inline clients/venice-admin-tool/src/main/java/com/linkedin/venice/datarecovery/DataRecoveryTask.java:22]: Can you add some more details what this means and why we need it?

Comment (lluwm) [reply clients/venice-admin-tool/src/main/java/com/linkedin/venice/datarecovery/DataRecoveryTask.java:22]: Thanks Nisarg for the review! The change description part probably has more details and explains why. Do you mean to move some of them here as comments?

Comment (nisargthakkar) [reply clients/venice-admin-tool/src/main/java/com/linkedin/venice/datarecovery/DataRecoveryTask.java:22]: Yeah, as javadoc or comments

Comment (lluwm) [reply clients/venice-admin-tool/src/main/java/com/linkedin/venice/datarecovery/DataRecoveryTask.java:22]: Sure, will add the comments and rename function to make it less confusing as we discussed offline.

**Review by nisargthakkar — APPROVED** [review]
LGTM

**Pull request merged**